common/src/main/java/org/opensearch/ml/common/memorycontainer/SemanticStorageConfig.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.memorycontainer;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ml.common.memorycontainer.MemoryContainerConstants.DIMENSION_FIELD;
+import static org.opensearch.ml.common.memorycontainer.MemoryContainerConstants.MODEL_ID_FIELD;
+import static org.opensearch.ml.common.memorycontainer.MemoryContainerConstants.MODEL_TYPE_FIELD;
+import static org.opensearch.ml.common.memorycontainer.MemoryContainerConstants.SEMANTIC_STORAGE_ENABLED_FIELD;
+
+import java.io.IOException;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.ml.common.FunctionName;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * Configuration for semantic storage in memory containers
+ */
+@Getter
+@Setter
+@Builder
+@EqualsAndHashCode
+public class SemanticStorageConfig implements ToXContentObject, Writeable {
+
+    private boolean semanticStorageEnabled;
+    private FunctionName modelType;
+    private String modelId;
+    private Integer dimension;
+
+    public SemanticStorageConfig(boolean semanticStorageEnabled, FunctionName modelType, String modelId, Integer dimension) {
+        this.semanticStorageEnabled = semanticStorageEnabled;
```

Comment:
Trying to understand the significance of this field. Let's say customer had false in this field but provided model and dimension. What will we do?

I think better to have a field to identify if we should create a k-nn index or regular index?